labgrid/driver/shelldriver.py

```diff
@@ -190,9 +190,6 @@ def _put_ssh_key(self, key):
             self.logger.debug("Read Key: %s", new_key)
             auth_keys, _, exitcode = self._run("cat ~/.ssh/authorized_keys")
             self.logger.debug("Exitcode: %s", exitcode)
-            if exitcode != 0:
-                self._run("mkdir ~/.ssh")
```

Comment:
We should still try to run mkdir -p ~/.ssh, so it works more often or RW root filesystems.
If it fails, return a message which explains that this directory needs to exist.